addons/spreadsheet/static/src/pivot/pivot_model.js

```diff
@@ -636,7 +639,223 @@ export class OdooPivotModel extends PivotModel {
             .map((dimension) => `${dimension.nameWithGranularity} ${dimension.order}`)
             .join(",");
         params.kwargs.order = order;
-        return super._getGroupsSubdivision(params, groupInfo);
+
+        const hasCustomField = allGroupBys.some((gb) => customFields[gb] !== undefined);
+        if (!hasCustomField) {
+            const result = await super._getGroupsSubdivision(params, groupInfo);
+            console.log("Custom group subdivision result:", result);
+            return result;
+        } else if (params.measureSpecs.some((measure) => measure.endsWith(":count_distinct"))) {
+            throw new Error(_t('Cannot use custom pivot groups with "Count Distinct" measure'));
+        } else {
+            return this._doCustomGroupSubdivision(params, groupInfo);
```

Comment:
The whole process is velu, not easy to read/follow/understand (the business is complex so it's ok), but it would be helpful to have an explanation of this process somewhere. For exemple, the paté in web/pivot_model already help me a lot in the past